packages/headless/src/api/platform-client-error-response.ts

```diff
@@ -0,0 +1,46 @@
+import {DisconnectedError} from '../utils/errors';
+
+export interface PlatformAPIErrorWithStatusCode {
+  statusCode: number;
+  message: string;
+  type: string;
```

Comment:
There's a lot of duplication between this file, and [this one](https://github.com/coveo/ui-kit/pull/3022/files#diff-d23883d1c9e76f00ba1cd5375015ac95f75215fabd518080b290fcd9ff541463R1). Why is it needed? Is it just due to the `type` parameter?

Reply:
Yep I should refactor that.